core/src/main/java/co/cask/wrangler/statistics/AddressFinder.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.wrangler.statistics;
+
+import au.com.bytecode.opencsv.CSVReader;
+
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static org.apache.commons.lang.math.NumberUtils.isNumber;
+
+/**
+ * Street address detector
+ * Recognize address components like city, state, street name, street number etc.
+ * 1905 N Lincoln Ave Apt 125 Urbana IL 61801
+ */
+public class AddressFinder {
+
+  //not in use because address parser is not accurate
```

Comment:
The comment is in the wrong place. It should be in AddressParser class, and AddressFinder shouldn't keep a reference to an AddressParser. AddressParser uses the regex patterns in AddressFinder to parse address, but for now AddressParser is not in use, because we don't parse an address anywhere. In the BasicStatistics, it calls AddressFinder.isUSAddress() to detect an address, only by matching street suffix keywords like "Street", "St." , "Road", "Rd".